containers/src/Kokkos_DynRankView.hpp

```diff
@@ -119,7 +119,7 @@ struct DynRankDimTraits {
       (std::is_same_v<Layout, Kokkos::LayoutRight> ||
        std::is_same_v<Layout, Kokkos::LayoutLeft>),
       Layout>
-  createLayout(const Layout& layout) {
+  createLayout(const Layout& layout, int new_rank = -1) {
```

Comment:
Does `new_rank` here need the `[[maybe_unused]]` markup as well? Looks like it is only used when L149 evaluates to true, otherwise the usage is compiled out?